controller_interface/include/controller_interface/helpers.hpp

```diff
@@ -78,6 +78,16 @@ inline bool interface_list_contains_interface_type(
          interface_type_list.end();
 }
 
+template <typename T>
+void add_element_to_list(std::vector<T> & list, const T & element)
+{
+  if (std::find(list.begin(), list.end(), element) == list.end())
+  {
+    // Only add to the list if it doesn't exist
+    list.push_back(element);
```

Comment:
when are we adding things multiple times?

Reply:
https://github.com/ros-controls/ros2_control/blob/a6e44abcfdc9a1246ac1c3e0d0e0091226f67953/controller_interface/src/chainable_controller_interface.cpp#L208-L232

here we are creating the interfaces from the `exported_state_interface_names_` and `exported_reference_interface_names_` and later we are readding again to the same vector in the method. I think it is better to check and insert to not to have inconsistent sizes.

This will only happen if we use these variables to create and export the interfaces

Reply:
sounds good, thank you